bosswave.cpp

```diff
@@ -7,6 +7,50 @@
 
 #include <msgpack.h>
 
+#include <cstdio>
+
+class NotImplementedException : public std::exception
```

Comment:
double check that exceptions work on all platforms before you commit to using them. I have a bad feeling that they might not.